src/common/js_logger.hpp

```diff
@@ -0,0 +1,130 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+#include <iostream>
+#include "sync/sync_manager.hpp"
+
+#if REALM_ANDROID
+    #include <android/log.h>
+#endif
+
+namespace realm {
+namespace common {
```

Comment:
Don't know :/, now that you mention I think the ``js_`` part in ``js_logger.hpp`` is wrong because that class doesn't have any JS behaviour inside, it just a logging delegator. 

We can call it just ``logger.hpp`` and maintain the namespace, so we can put there classes that provide similar services (reusable classes that encapsulate things that has nothing to do with JS but are consumed by JS classes) ?

Reply:
Good point calling it `logger.hpp`. Since we only have one file in the `realm::common`, I don't think we should create a new folder for it.